### PR TITLE
std: Add process_vm_readv/writev wrappers

### PR DESCRIPTION
**Pull request by chivay**
Hi,
this patch adds wrappers for process_vm_readv/writev syscalls on Linux.

### CODE REVIEW

**Review by ifreund — COMMENTED** [review]
Why is flags a `u64`? My man page says it is an `unsigned long` and thus 32 bits on `i386` right? It seems to me that it should be a `usize` like the rest.

**Comment by chivay** [comment]
>Why is flags a u64? My man page says it is an unsigned long and thus 32 bits on i386 right? It seems to me that it should be a usize like the rest.

Yes, you're right. It should be fixed now.

File: lib/std/os/linux.zig [hunk 1]
```diff
@@ -1410,6 +1410,32 @@ pub fn pidfd_send_signal(pidfd: fd_t, sig: i32, info: ?*siginfo_t, flags: u32) u
...
+pub fn process_vm_readv(pid: pid_t, local: [*]const iovec, local_count: usize, remote: [*]const iovec, remote_count: usize, flags: usize) usize
+{
+    return syscall6(
+        .process_vm_readv,
+        @bitCast(usize, @as(isize, pid)),
+        @ptrToInt(local),
+        local_count,
+        @ptrToInt(remote),
+        remote_count,
+        flags
```
Comment (ifreund) [inline lib/std/os/linux.zig:1422]: without the trailing comma, `zig fmt` will put this all on one line.
```suggestion
        flags,
```

File: lib/std/os/linux.zig [hunk 2]
```diff
@@ -1410,6 +1410,32 @@ pub fn pidfd_send_signal(pidfd: fd_t, sig: i32, info: ?*siginfo_t, flags: u32) u
...
+pub fn process_vm_writev(pid: pid_t, local: [*]const iovec, local_count: usize, remote: [*]const iovec, remote_count: usize, flags: usize) usize
+{
+    return syscall6(
+        .process_vm_writev,
+        @bitCast(usize, @as(isize, pid)),
+        @ptrToInt(local),
+        local_count,
+        @ptrToInt(remote),
+        remote_count,
+        flags
```
Comment (ifreund) [inline lib/std/os/linux.zig:1435]:
```suggestion
        flags,
```

File: lib/std/os/linux.zig [hunk 3]
```diff
@@ -1410,6 +1410,32 @@ pub fn pidfd_send_signal(pidfd: fd_t, sig: i32, info: ?*siginfo_t, flags: u32) u
     );
 }
 
+pub fn process_vm_readv(pid: pid_t, local: [*]const iovec, local_count: usize, remote: [*]const iovec, remote_count: usize, flags: usize) usize
+{
```
Comment (ifreund) [inline lib/std/os/linux.zig:1414]: zig fmt wants the curly on the same line as the signature.
```suggestion
pub fn process_vm_readv(pid: pid_t, local: [*]const iovec, local_count: usize, remote: [*]const iovec, remote_count: usize, flags: usize) usize {
```

File: lib/std/os/linux.zig [hunk 4]
```diff
@@ -1410,6 +1410,32 @@ pub fn pidfd_send_signal(pidfd: fd_t, sig: i32, info: ?*siginfo_t, flags: u32) u
...
+        @ptrToInt(local),
+        local_count,
+        @ptrToInt(remote),
+        remote_count,
+        flags
+    );
+}
+
+pub fn process_vm_writev(pid: pid_t, local: [*]const iovec, local_count: usize, remote: [*]const iovec, remote_count: usize, flags: usize) usize
+{
```
Comment (ifreund) [inline lib/std/os/linux.zig:1427]:
```suggestion
pub fn process_vm_writev(pid: pid_t, local: [*]const iovec, local_count: usize, remote: [*]const iovec, remote_count: usize, flags: usize) usize {
```

**Comment by chivay** [comment]
Applied zig fmt

**Review by ifreund — APPROVED** [review]
LGTM, just waiting on CI now.

**Pull request merged**